describe.go

```diff
@@ -92,13 +152,61 @@ func Describe(message *Message, w io.Writer, filters ...FieldFilter) error {
 	return nil
 }
 
-func sortFieldIDs(fields map[int]field.Field) []int {
-	keys := make([]int, 0, len(fields))
+func sortFieldIDs(fields map[string]field.Field) []string {
+	numericKeys := make([]int, 0)
+	nonNumericKeys := make([]string, 0)
+
 	for k := range fields {
-		keys = append(keys, k)
+		if id, err := strconv.Atoi(k); err == nil {
+			numericKeys = append(numericKeys, id)
+		} else {
+			nonNumericKeys = append(nonNumericKeys, k)
+		}
 	}
 
-	sort.Ints(keys)
+	// Sorting numeric and non-numeric keys separately
+	sort.Ints(numericKeys)
+	sort.Strings(nonNumericKeys)
+
+	keys := make([]string, 0, len(fields))
+
+	// Appending numeric keys first (as strings)
+	for _, key := range numericKeys {
+		keys = append(keys, strconv.Itoa(key))
+	}
+
+	// Appending non-numeric keys
+	keys = append(keys, nonNumericKeys...)
 
 	return keys
 }
+
+// splitAndAnnotate splits bits blocks and annotates them with bit numbers
+// and splits them by 32 bits if needed
+func splitAndAnnotate(bits string) string {
+	bitBlocks := strings.Split(bits, " ")
+	if len(bitBlocks) == 0 {
```

Comment:
```suggestion
	bitBlocks := strings.Split(bits, " ")
	if len(bitBlocks) == 1 {
```

If the 2nd arg, `separator` of `Split` is not empty and the first arg `bits` does not contain the separator, the returned slice will have a length of 1 with one element being the input - according to the docs.